WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsDataSource.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+
+// MARK: - ReaderInterestViewModel
+class ReaderInterestViewModel {
+    var isSelected: Bool = false
+
+    var title: String {
+        return interest.title
+    }
+
+    var slug: String {
+        return interest.slug
+    }
+
+    private var interest: RemoteReaderInterest
+
+    init(interest: RemoteReaderInterest) {
+        self.interest = interest
+    }
+
+    public func toggleSelected() {
+        self.isSelected = !isSelected
+    }
+}
+
+// MARK: - ReaderInterestsDataDelegate
+protocol ReaderInterestsDataDelegate: AnyObject {
+    func readerInterestsDidUpdate(_ dataSource: ReaderInterestsDataSource)
+}
+
+protocol ReaderInterestsService: AnyObject {
+    func fetchInterests(success: @escaping ([RemoteReaderInterest]) -> Void,
+                        failure: @escaping (Error) -> Void)
+}
+
+// MARK: - ReaderInterestsDataSource
+class ReaderInterestsDataSource {
+    var delegate: ReaderInterestsDataDelegate?
+
+    private(set) var count: Int = 0
+    private(set) var interests: [ReaderInterestViewModel] = [] {
+        didSet {
+            count = interests.count
```

Comment:
What is this count used for?

Reply:
It keeps track of how many interests there are and is used in the `numberOfItemsInSection` method. 

The idea is that an external class doesn't access the interests array itself. 

To keep to that point I've made the interests array completely private to ensure the right methods are called.